inotify/adapters.py

```diff
@@ -147,11 +138,11 @@ def __handle_inotify_event(self, wd, event_type):
 
             filename = self.__buffer[_STRUCT_HEADER_LENGTH:event_length]
             # Our filename is 16-byte aligned and right-padded with NULs.
-            filename = self.to_unicode(filename).rstrip('\0')
+            filename = fsdecode(filename).rstrip('\0')
 
             self.__buffer = self.__buffer[event_length:]
 
-            path = self.to_unicode(self.__watches_r.get(header.wd))
+            path = fsencode(self.__watches_r.get(header.wd))
```

Comment:
Previously requested that you just index this and catch it being a None prior to making the call.